homeassistant/components/tplink/strings.json

```diff
@@ -116,6 +143,36 @@
       },
       "available_firmware_version": {
         "name": "Available firmware version"
+      },
+      "battery_level": {
+        "name": "Battery level"
+      },
+      "temperature": {
+        "name": "[%key:component::sensor::entity_component::temperature::name%]"
+      },
+      "voltage": {
+        "name": "[%key:component::sensor::entity_component::voltage::name%]"
+      },
+      "current": {
+        "name": "[%key:component::sensor::entity_component::current::name%]"
+      },
+      "humidity": {
+        "name": "[%key:component::sensor::entity_component::humidity::name%]"
+      },
+      "device_time": {
+        "name": "Device time"
+      },
+      "auto_off_at": {
+        "name": "Auto off at"
+      },
```

Comment:
I'm not sure, I don't *think* it would have been working unless your entity registry retained some `feature.name` values from when we were dynamically creating some entities. The entity registry entry from the snapshot for `device_time` prior to this commit was:
```diff
# name: test_states[sensor.my_device_timestamp_2-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': <RegistryEntryDisabler.INTEGRATION: 'integration'>,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
+   'entity_id': 'sensor.my_device_timestamp_2',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.TIMESTAMP: 'timestamp'>,
    'original_icon': None,
+   'original_name': 'Timestamp',
    'platform': 'tplink',
    'previous_unique_id': None,
    'supported_features': 0,
+   'translation_key': 'device_time',
    'unique_id': '123456789ABCDEFGH_device_time',
    'unit_of_measurement': None,
  })
  ```
There's no state for `device_time` as it's disabled but for `auto_off_at`:
```diff
# name: test_states[sensor.my_device_timestamp-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'timestamp',
+     'friendly_name': 'my_device Timestamp',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.my_device_timestamp',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '2024-06-24T09:03:11+00:00',
  })
# ---
```